src/main/java/com/dev/cinema/service/impl/MovieSessionServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package com.dev.cinema.service.impl;
+
+import com.dev.cinema.dao.MovieSessionDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.models.MovieSession;
+import com.dev.cinema.service.MovieSessionService;
+import com.dev.cinema.util.HibernateUtil;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import org.hibernate.Session;
+
+@Service
+public class MovieSessionServiceImpl implements MovieSessionService {
+    @Inject
+    private MovieSessionDao movieSessionDao;
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
any suggestion what can be wrong with this method ?

Reply:
it supposed be at DAO layer?

Reply:
sure